dev/scaletesting/bulkreposettings/main.go

```diff
@@ -145,22 +201,118 @@ var app = &cli.App{
 	},
 }
 
-func fetchRepos(ctx context.Context, org string, gh *github.Client) ([]*store.Repo, error) {
+type Iter[T any] interface {
+	Err() error
+	Next(ctx context.Context) T
+	Done() bool
+}
+
+var _ Iter[[]*store.Repo] = (*GithubRepoFetcher)(nil)
+
+// StaticRepoFetcher satisfies the Iter interface allowing one to iterate over a static array of repos. To change
+// how many repos are returned per invocation of next, set iterSize (default 10). To start iterating at a different
+// index, set start to a different value.
+//
+// The iteration is considered done when start >= len(repos)
+type StaticRepoFetcher struct {
```

Comment:
nit: Usually, we're naming those things `Mock*` unless we're actually using them outside the tests.

```suggestion
type MockRepoFetcher struct {
```